packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,372 @@
+
+    this.update((state: NetworkEnablementControllerState) => {
+      if (namespace && state.enabledNetworkMap[namespace]) {
+        state.enabledNetworkMap[namespace][networkId] = false;
+      }
+    });
+  }
+
+  /**
+   * Checks if a network is enabled.
+   *
+   * @param chainId - The chain ID of the network to check.
+   * @returns True if the network is enabled, false otherwise.
+   */
+  isNetworkEnabled(chainId: Hex | CaipChainId): boolean {
+    let namespace: CaipNamespace | undefined;
+    let networkId: string;
+
+    if (!isHexString(chainId) && !isCaipChainId(chainId)) {
+      return false;
+    }
+
+    // Handle both Hex and CaipChainId formats
+    if (isHexString(chainId)) {
+      const caipChainId = toEvmCaipChainId(chainId as Hex);
+      const parsed = parseCaipChainId(caipChainId);
+      namespace = parsed.namespace;
+      networkId = chainId;
+    } else {
+      const parsed = parseCaipChainId(chainId as CaipChainId);
+      namespace = parsed.namespace;
+      networkId = chainId as CaipChainId;
+    }
+
+    if (!namespace) {
+      return false;
+    }
+
+    return this.state.enabledNetworkMap[namespace]?.[networkId] === true;
+  }
+
+  /**
+   * Sets the enabled networks in the controller state.
+   * This method updates the enabledNetworkMap to mark specified networks as enabled.
+   * It can handle both a single chain ID or an array of chain IDs.
+   *
+   * @param chainIds - A single CAIP-2 chain ID (e.g. 'eip155:1') or an array of chain IDs
+   * to be enabled. All other networks will be implicitly disabled.
+   */
+  setEnabledNetworks(chainIds: CaipChainId | CaipChainId[]) {
+    const ids = Array.isArray(chainIds) ? chainIds : [chainIds];
+
+    // Clear all existing enabled networks first
+    this.update((state: NetworkEnablementControllerState) => {
+      Object.keys(state.enabledNetworkMap).forEach((namespace) => {
+        state.enabledNetworkMap[namespace as CaipNamespace] = {};
+      });
+    });
+
+    // Enable the specified networks
+    ids.forEach((chainId) => {
+      const { namespace, reference } = parseCaipChainId(chainId);
+      if (namespace === KnownCaipNamespace.Eip155) {
+        // For EVM chains, convert the chain ID to hex format
+        const hexChainId = toHex(parseInt(reference, 10));
+        this.setEvmEnabledNetwork(hexChainId);
+      } else if (namespace === KnownCaipNamespace.Solana) {
+        this.setSolanaEnabledNetwork(chainId);
+      }
+    });
+  }
+
+  /**
+   * Gets all enabled networks for a specific namespace.
+   *
+   * @param namespace - The namespace to get enabled networks for.
+   * @returns Array of enabled network IDs for the namespace.
+   */
+  getEnabledNetworksForNamespace(namespace: CaipNamespace): string[] {
+    const enabledNetworks = this.state.enabledNetworkMap[namespace];
+    if (!enabledNetworks) {
+      return [];
+    }
+    return Object.keys(enabledNetworks).filter(
+      (networkId) => enabledNetworks[networkId] === true,
```

Comment:
Nit: Wouldn't this by default be a boolean ?
```suggestion
      (networkId) => enabledNetworks[networkId],